Adding-a-new-device.md

```diff
@@ -0,0 +1,224 @@
+## Adding a new device
+
+### Running the gateway
+
+Run the gateway using the the -d option.
+
+The following run.sh script makes a backup of the addon directory (I've accidentally wiped it out before) and then starts the gateway with the
+`-d` options and runs the output through a sed script which filters the color escape sequences. The sed part isn't really needed anymore since
+we now have persistent logging.
+
+Typical run.sh
+```
+#!/bin/bash
+
+#set -x
+
+BACKUP_ROOT="${HOME}/addon-backup"
+BACKUP_DIR="${BACKUP_ROOT}/$(date +%Y-%m-%d-%H%M%S)"
+SRC_DIR="${HOME}/moziot/.mozilla-iot/addons"
+
+# Remove all but the last 20 directories
+ls -1t ${BACKUP_ROOT} | tail -n +20 | while read dir; do
+  echo "Removing ${BACKUP_ROOT}/${dir} ..."
+  rm -rf ${BACKUP_ROOT}/${dir}
+done
+
+# Backup the current directory
+mkdir -p ${BACKUP_DIR}
+echo "Backing up ${SRC_DIR} to ${BACKUP_DIR} ..."
+rsync -ap --exclude=node_modules ${SRC_DIR}/ ${BACKUP_DIR}/
+
+# Start the gateway
+npm start -- -d 2>&1 | tee >(sed -r 's/\x1B\[[0-9]{1,2}m//g' > zb.log)
+```
+
+### Pair the device
+
+Pair the device. It will probably show up as a custom thing with no properties or perhaps a battery level.
+
+### Run the CLI
+
+Run the CLI: https://github.com/mozilla-iot/cli
+
+```
+cd cli
+./c.py gateway http://localhost:8080
+```
+
+The `devices` command will show the detected devices. Disabling all of the adapters except for the zwave one will reduce the amount of log noise. Having a minimal number of other devices paired also helps.
+
+```
+cli http://localhost:8080> devices
```

Comment:
cli -> ./c.py

Also, stray > ?

Reply:
That's the prompt. The > is the end of the prompt and devices is the part that you type.